webpack/assets/javascripts/react_app/components/ConfirmModal/slice.js

```diff
@@ -0,0 +1,51 @@
+import Immutable from 'seamless-immutable';
+import { noop } from '../../common/helpers';
+
+export const OPEN_CONFIRM_MODAL = 'OPEN_CONFIRM_MODAL';
+export const CLOSE_CONFIRM_MODAL = 'CLOSE_CONFIRM_MODAL';
+
+// actions
+export const openConfirmModal = ({
+  title = '',
+  message = '',
+  onConfirm = noop,
+  onCancel = noop,
+  isWarning = false,
+  confirmButtonText = null,
+  modalProps = {},
+}) => ({
+  type: OPEN_CONFIRM_MODAL,
+  payload: {
+    title,
+    message,
+    onConfirm,
+    onCancel,
+    modalProps,
+    isWarning,
+    confirmButtonText,
+  },
+});
+
+export const closeConfirmModal = () => ({
+  type: CLOSE_CONFIRM_MODAL,
+  payload: {},
+});
```

Comment:
this can be a simple action
```suggestion
export const closeConfirmModal = ({
  type: CLOSE_CONFIRM_MODAL,
  payload: {},
});
```

Reply:
nice! didn't think about it :)

Reply:
but then again... I think the convention for actions is a function, and people will try to call it `closeConfirmModal()`.

Reply:
it's not a convention for actions by redux, just for thunked actions (mainly for asynchronous needs)